apps/site/assets/ts/helpers/service.ts

```diff
@@ -103,44 +75,46 @@ export const groupServiceByDate = (
   return [
     {
       type: "future",
-      servicePeriod: `starts ${formattedDate(startDateObject)}`,
+      servicePeriod: `${formattedDate(startDateObject)} to ${formattedDate(
+        endDateObject
+      )}`,
       service
     }
   ];
 };
 
-export const serviceDate = (
-  service: Service | ServiceWithServiceDate
-): string => {
-  const { start_date: startDate, end_date: endDate } = service;
-  if (service.start_date === service.end_date) {
-    return holidayDate(service);
-  }
+export const optGroupNames: ServiceOptGroupName[] = [
+  "current",
+  "holiday",
+  "future"
+];
 
-  const startDateObject = new Date(startDate);
-  const endDateObject = new Date(endDate);
-  if ("service_date" in service) {
-    // Get unix timestamps
-    const serviceDateTime = new Date(service.service_date).getTime();
-    const startDateTime = startDateObject.getTime();
-    const endDateTime = endDateObject.getTime();
-
-    if (serviceDateTime >= startDateTime && serviceDateTime <= endDateTime) {
-      return `ends ${formattedDate(endDateObject)}`;
-    }
+const optGroupTitles: { [key in ServiceOptGroupName]: string } = {
+  current: "Current Schedules",
+  holiday: "Holidays",
+  future: "Future Schedules"
+};
 
-    if (serviceDateTime < startDateTime) {
-      return `starts ${formattedDate(startDateObject)}`;
+export const groupedServiceOptLabel = (
+  groupName: ServiceOptGroupName,
+  groupedServices: ServiceByOptGroup[]
+): string => {
+  // TODO: Validate whether all services in a group will have the same rating description and service period. Also check correspondence between service period and rating start/end dates.
```

Comment:
Could we group the services based on a combination of rating description + service period? Then we wouldn't need to make this assumption, and maybe the service picker would automatically handle a wider range of unusual situations.